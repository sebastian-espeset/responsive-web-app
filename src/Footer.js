import React from 'react';
import { StyledFooterContainer, StyledFooterImageContainer, StyledFooterTextContainer } from './styles/Footer.style';
import logo from './assets/logo.png';
// small typo found in figma re: "Im 'a' Employer"
export default function Footer() {
  return (
    <StyledFooterContainer>
      <StyledFooterImageContainer>
        <img src={logo} alt="beam logo" data-testid='beam-logo-footer-test-id'/>
      </StyledFooterImageContainer>
      <StyledFooterTextContainer>
        <h3>Explore</h3>
        <h4>I'm a Broker</h4>
        <h4>I'm a Member</h4>
        <h4>I'm an Employer</h4>
        <h4>I'm a Provider</h4>
      </StyledFooterTextContainer>
      <StyledFooterTextContainer>
        <h3>Learn</h3>
        <h4>FAQs</h4>
        <h4>Company Blog</h4>
        <h4>Product Blog</h4>
      </StyledFooterTextContainer>
      <StyledFooterTextContainer>
        <h3>Company</h3>
        <h4>About</h4>
        <h4>Careers</h4>
        <h4>Terms</h4>
        <h4>Privacy</h4>
      </StyledFooterTextContainer>
      <StyledFooterTextContainer>
        <h3>Resources</h3>
        <h4>Find a Dentist</h4>
        <h4>Contact Us</h4>
        <h4>Get a Quote</h4>
        <h4>Paper Claims</h4>
      </StyledFooterTextContainer>
      <StyledFooterTextContainer>
        <h3>Connect</h3>
        <h4>LinkedIn</h4>
        <h4>Twitter</h4>
        <h4>Facebook</h4>
        <h4>Instagram</h4>
      </StyledFooterTextContainer>
    </StyledFooterContainer>
  )
}
