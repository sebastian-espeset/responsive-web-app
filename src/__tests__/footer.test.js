import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

afterEach(()=>{
    cleanup();
});

test('footer section headline should render',()=>{
    //Arrange
    render(<Footer/>)
    //Act
    const footerBeamLogo = screen.getByTestId('beam-logo-footer-test-id');
    //Assert
    expect(footerBeamLogo).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('footer component to match snapshot',()=>{
    const tree = renderer.create(<Footer/>).toJSON();
    expect(tree).toMatchSnapshot();
})