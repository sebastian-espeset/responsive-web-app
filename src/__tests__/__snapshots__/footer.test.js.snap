// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer component to match snapshot 1`] = `
<div
  className="sc-bczRLJ gkjUam"
>
  <div
    className="sc-gsnTZi hLfmjy"
  >
    <img
      alt="beam logo"
      data-testid="beam-logo-footer-test-id"
      src="logo.png"
    />
  </div>
  <div
    className="sc-dkzDqf evhUSi"
  >
    <h3>
      Explore
    </h3>
    <h4>
      I'm a Broker
    </h4>
    <h4>
      I'm a Member
    </h4>
    <h4>
      I'm an Employer
    </h4>
    <h4>
      I'm a Provider
    </h4>
  </div>
  <div
    className="sc-dkzDqf evhUSi"
  >
    <h3>
      Learn
    </h3>
    <h4>
      FAQs
    </h4>
    <h4>
      Company Blog
    </h4>
    <h4>
      Product Blog
    </h4>
  </div>
  <div
    className="sc-dkzDqf evhUSi"
  >
    <h3>
      Company
    </h3>
    <h4>
      About
    </h4>
    <h4>
      Careers
    </h4>
    <h4>
      Terms
    </h4>
    <h4>
      Privacy
    </h4>
  </div>
  <div
    className="sc-dkzDqf evhUSi"
  >
    <h3>
      Resources
    </h3>
    <h4>
      Find a Dentist
    </h4>
    <h4>
      Contact Us
    </h4>
    <h4>
      Get a Quote
    </h4>
    <h4>
      Paper Claims
    </h4>
  </div>
  <div
    className="sc-dkzDqf evhUSi"
  >
    <h3>
      Connect
    </h3>
    <h4>
      LinkedIn
    </h4>
    <h4>
      Twitter
    </h4>
    <h4>
      Facebook
    </h4>
    <h4>
      Instagram
    </h4>
  </div>
</div>
`;
